server/api_device_stats.go

```diff
@@ -40,13 +40,34 @@ type DeviceStats struct {
 	ImpactRates  [2016]float64
 }
 
+//  DeviceStatsFriendlyJSON is strictly for JSON encoding
+// It is a friendly version of DeviceStats that includes the PublicKey as a hex string
+// and also includes the ShortId of the device to avoid having to look it up in another HTTP request
+// It contains pointers to the arrays to avoid copying the data
+// DeviceStatsFriendlyJSON's lifetime is limited to the HTTP request by design
+type DeviceStatsFriendlyJSON struct {
+	PublicKey    string
```

Comment:
This should be a `glow.PublicKey`, then you have it encoded as hex in the `MarshalJSON()` function

Reply:
that gives you type safety in the go code while still giving out the output you want in the json. You already made a `MarshalJSON()` function, might as well use it